### ISSUE TITLE
Ajouter un champ textuel pour les prédicats utilisés par le script

### ISSUE THREAD

**Issue by Ulty** [stephaned68/COF2e#12]
Dans un premier emps, j'aurais besoin d'un champ textuel sous l'onglet Script, pour mettre les prédicats du script.

À terme, j'aimerais avoir un tel champ associé 

- aux armes (pas besoin pour les sorts et attaques naturelles)
- aux armures et autres équipements
- aux capacités (uniquement visibles en mode édition)

**Comment by stephaned68** [comment]
Celui de l'onglet script est déjà en place. Crées un attribut scriptVersion à true et scriptVersion_max avec le niveau de version du script et le contenu de l'onglet apparaîtra avec un textarea pour les prédicats

**Comment by Ulty** [comment]
J'ai une fiche sous les yeux, avec les bons attributs. Je vois l'onglet Script, mais il n'y a rien dedans.

**Comment by Ulty** [comment]
![image](https://github.com/user-attachments/assets/793a6f09-1d1a-4617-ae01-3bbbc0c3b5a1)

**Comment by stephaned68** [comment]
J'ai pourtant repris le nom de l'attribut de la fiche COF1, il me semble...

![image](https://github.com/user-attachments/assets/45f337c6-541d-4313-9d54-0d1600705915)

![image](https://github.com/user-attachments/assets/44f55d6d-2c57-4047-82d1-cc25ffdd8903)

**Comment by Ulty** [comment]
Voici l'attribut dans la fiche :
![image](https://github.com/user-attachments/assets/c640db66-742f-45de-97cb-64e3ae29cb04)

**Comment by Ulty** [comment]
On devrait pouvoir utiliser un nom d'attribut entièrement en minuscule, peut-être, non ?

**Comment by stephaned68** [comment]
La version de la fiche est 1.1.0 ?

**Comment by Ulty** [comment]
Non, c'était 1.0.0. Bizare, ça n'avait pas fait la mise à jour vers la nouvelle version. Je viens de la forcer, et ça marche.

**Comment by stephaned68** [comment]
Euh ça n'a toujours pas été accepté par Roll20, donc pour avoir mes derniers devs il faut choper le code depuis mon Github dans une partie custom...

**Comment by Ulty** [comment]
Oui, c'est pourtant ce que j'avais fait. J'utilise le bac à sable de fiches.
Désolé pour le dérangement...

**Comment by Ulty** [comment]
Je pense qu'on peut fermer cette issue, alors. J'en ouvrirai une pour les armes quand il y aura les attributs d'arme en main, et une pour les équipement quand il y aura la gestion des équipements portés. Pour les capacités, ça peut vraiment attendre.

**Comment by stephaned68** [comment]
Pour les capacités j'ai commencé à implémenter une reconnaissance de certaines particularités comme Peau de Pierre du barbare qui change le calcul de la DEF si la CON est supérieure à l'AGI. On peut sans doute partir du framework que j'ai mis en place pour ça...

**Comment by Ulty** [comment]
Ça consiste en quoi, ce framework ?

**Comment by stephaned68** [comment]
Un attribut caché capabilities qui contient la liste des noms des capacités avec traitement spécial possédées par le personnage, sans espace, en lowercase, délimités par des virgules. Pour l'instant on ne gère que "peaudepierre". En y repensant ça fait double emploi avec les prédicats de l'onglet script...

**Comment by Ulty** [comment]
L'idée des prédicats en COF2, ce sera de mettre tout ce qui n'est pas dans les capacités normales (qui suivent les règles), donc c'est bien d'avoir un second attribut pour les capacités. Au final, j'irai chercher toutes les capacités cochées, car je vais essayer d'implémenter un support pour toute. C'est l'objectif, en tout cas.
